extensions/hibernate-search-orm-elasticsearch/runtime/src/main/java/io/quarkus/hibernate/search/orm/elasticsearch/runtime/devconsole/HibernateSearchDevConsoleRecorder.java

```diff
@@ -31,19 +32,24 @@ protected void handlePostAsync(RoutingContext event, MultiMap form) throws Excep
                 if (form.isEmpty()) {
                     return;
                 }
-                SearchMapping mapping = HibernateSearchSupplier.searchMapping();
-                if (mapping == null) {
+                Map<String, SearchMapping> mappings = HibernateSearchSupplier.searchMapping();
+                if (mappings.isEmpty()) {
                     flashMessage(event, "There are no indexed entity types.", FlashScopeUtil.FlashMessageStatus.ERROR);
                     return;
                 }
-                mapping.scope(Object.class,
-                        mapping.allIndexedEntities().stream()
-                                .map(SearchIndexedEntity::jpaName)
-                                .filter(form::contains)
-                                .collect(Collectors.toList()))
-                        .massIndexer()
-                        .startAndWait();
-                flashMessage(event, "Entities successfully reindexed", Duration.ofSeconds(10));
+                for (SearchMapping mapping : mappings.values()) {
+                    List<String> entityNames = mapping.allIndexedEntities().stream()
+                            .map(SearchIndexedEntity::jpaName)
+                            .filter(form::contains)
+                            .collect(Collectors.toList());
+                    if (!entityNames.isEmpty()) {
+                        mapping.scope(Object.class,
+                                entityNames)
+                                .massIndexer()
+                                .startAndWait();
+                        flashMessage(event, "Entities successfully reindexed", Duration.ofSeconds(10));
+                    }
```

Comment:
This will produce unwanted side effects, as selecting an entity with a given name in a given persistence unit in the UI, will reindex all entities with that name in all persistence units.

You should avoid that.

I would suggest moving the "reindex" button next to the name of each persistence unit in the UI, and passing the persistence unit name along with the entity names when calling this REST service, so that you know precisely which `SearchMapping` to use for reindexing.

Reply:
left a single reindex button because it seems like a better user experience to me, but updated the form to send the associated persistence unit as value along with the selected jpaEntity name, so that we can find the relevant searchMapping